CONTRIBUTING.md

```diff
@@ -0,0 +1,192 @@
+            }
+        }
+      ]
+    }
+    ```
+
+1. Navigate to `azure-functions-host/src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs` class and add breakpoint at `_process.Start()`.
+
+   >> Note: Make sure that WebJobs.Script.WebHost is set as a Startup Project
+
+1. Now in another IDE instance, git clone your fork for [azure function worker](https://github.com/helayoty/azure-functions-java-worker) and open it.
+
+1. Add new remote debugging configuration. 
+
+  Here is a sample configuration for IntelliJ IDEA 
+
+![IntelliJ IDEA Remote Configuration](tools/.images/worker-debug-configuration.png)
+
+![IntelliJ IDEA Config Values](tools/.images/remote-config.png)
+
+With this you have completed the setup for Azure Functions Host and the Azure Functions Java Worker Development.
+
+### Start Debugging
+
+1. Run the azure-functions-host from Step 5 in debugging mode (.NET Core Launch Debug). Wait until it stops at the `_process.Start()` breakpoint. At this point, the host is started and the java language worker needs to connect to the host.
+
+1. Navigate to `azure-functions-java-worker/src/main/java/com/microsoft/azure/functions/worker/JavaWorkerClient.java` and add a breakpoint in the constructor. Click the Remote Debugging configured in Step 7. This connects the worker process to the host and runs worker in debugging mode.
+
+1. Run through the breakpoint in the host until it hits the breakpoint in `JavaWorkerClient.java`. At this point the worker and host are up and running with your function app (this was configured as part of the environment variables above)
+
+1. Open a browser and invoke the HTTP Trigger and it will reach the `execute` method in `azure-functions-java-worker/src/main/java/com/microsoft/azure/functions/worker/broker/JavaMethodExecutorImpl.java` and can then follow along the execution of the function from here.
+
+### Testing a new version of the Java Worker
+
+- In case you need to test a new version from Java worker code, you'll need to:
+
+  - Compile the worker `mvn clean package -Dmaven.javadoc.skip=true -Dmaven.test.skip -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B`
+
+  - Navigate to target folder and rename `azure-functions-java-worker-1.8.0.jar` to `azure-functions-java-worker.jar`
+
+  - Copy the jar file to the debug folder in the host code base(<Azure_Function_Host_Path>/azure-functions-host/src/WebJobs.Script.WebHost/bin/Debug/netcoreapp3.1/workers/java)
+
+  - Restart host debugger
+
+### Running unit tests
+
+1. Add your unit tests under ./src/test folder
+2. Run: `mvn clean package`
+
+## Continuous Integration Guidelines & Conventions
+
+This project uses a combination of Azure DevOps and GitHub Actions for CI/CD.
+
+- For each PR request/merge, a continuous integration pipeline will run internally that performs linting and running unit tests on your PR/merge.
```

Comment:
Same as above. Non-Forked user's PR will run automatically, forked repo user need to ask us to run the pipeline.